libtenzir/src/tql2/tokens.cpp

```diff
@@ -57,14 +57,20 @@ auto tokenize_permissive(std::string_view content) -> std::vector<token> {
       ->* [] { return tk::datetime; }
     | ignore(digit >> *digit_us >> -('.' >> digit >> *digit_us) >> -identifier)
       ->* [] { return tk::scalar; }
+    | ignore("b\"" >> *(('\\' >> any) | (any - '"')) >> '"')
+      ->* [] { return tk::blob; }
     | ignore('"' >> *(('\\' >> any) | (any - '"')) >> '"')
       ->* [] { return tk::string; }
     | ignore('"' >> *(('\\' >> any) | (any - '"')))
       ->* [] { return tk::error; } // non-terminated string
+    | ignore("br\"" >> *(any - '"') >> '"')
+      ->* [] { return tk::raw_blob; }
     | ignore("r\"" >> *(any - '"') >> '"')
       ->* [] { return tk::raw_string; }
     | ignore("r\"" >> *(any - '"'))
       ->* [] { return tk::error; } // non-terminated raw string
+    | ignore("br#\"" >> *(any - "\"#") >> "\"#")
+      ->* [] { return tk::raw_blob; }
```

Comment:
The blob tokens are missing the non-terminated version. I could not come up with an example where the program is still accepted, but basically `b"…` can get tokenized as `b` and then `"…` with these rules, but you want to lock into reading a blob literal when reading the `b"` prefix, and produce an error if you don't succeed, without falling back to tokenizing `b` as an identifier.

Reply:
@dominiklohmann Pushed a commit on your branch as requested. I also noticed that this can be much simpler, and that we should catch also things like `xyz"…"` to reserve all string prefixes.

Reply:
Great, thanks for doing this directly. 🙏